lnd_test.go

```diff
@@ -4122,6 +4122,85 @@ func testSendToRouteErrorPropagation(net *lntest.NetworkHarness, t *harnessTest)
 	closeChannelAndAssert(ctxt, t, net, carol, chanPointCarol, false)
 }
 
+// check unannounced channels are not returned by describeGraph RPC request
+// unless explicity asked for.
+func testUnannouncedChannels(net *lntest.NetworkHarness, t *harnessTest) {
+	timeout := time.Duration(time.Second * 5)
+
+	// Open a channel between Alice and Bob, ensuring the
+	// channel has been opened properly.
+	amount := maxBtcFundingAmount
+	ctx, _ := context.WithTimeout(context.Background(), timeout)
+
+	chanOpenUpdate, err := net.OpenChannel(
+		ctx, net.Alice, net.Bob,
+		lntest.OpenChannelParams{
+			Amt: amount,
+		},
+	)
+	if err != nil {
+		t.Fatalf("unable to open channel: %v", err)
+	}
+
+	// Mine 2 blocks, and check that the channel is opened but not yet
+	// announced to the network
+	mineBlocks(t, net, 2)
+
+	// figure out the chanPoint, required to close the channel
+	fundingChanPoint, err := net.WaitForChannelOpen(ctx, chanOpenUpdate)
+	if err != nil {
+		t.Fatalf("error while waiting for channel open: %v", err)
+	}
+
+	// Alice should have 1 edge in her graph.
+	req := &lnrpc.ChannelGraphRequest{IncludeUnannounced: true}
+	chanGraph, err := net.Alice.DescribeGraph(ctx, req)
+	if err != nil {
+		t.Fatalf("unable to query alice's graph: %v", err)
+	}
+
+	numEdges := len(chanGraph.Edges)
+	if numEdges != 1 {
+		t.Fatalf("expected to find 1 edge in the graph, found %d", numEdges)
+	}
+
+	// But none already announced
+	req.IncludeUnannounced = false
+	chanGraph, err = net.Alice.DescribeGraph(ctx, req)
+	if err != nil {
+		t.Fatalf("unable to query alice's graph: %v", err)
+	}
+
+	numEdges = len(chanGraph.Edges)
+	if numEdges != 0 {
+		t.Fatalf("expected to find 0 announced edges in the graph, found %d", numEdges)
+	}
+
+	// Mine 4 more blocks, and check that the channel is now advertized
+	mineBlocks(t, net, 4)
+
+	// Give the network a chance to learn that auth proof is confirmed
+	time.Sleep(time.Millisecond * 50)
```

Comment:
Instead of sleeping here, you should be using a `WaitPredicate`. Check other tests for how this is done :)